File: server/BarAPP/src/main/java/org/pbarreiro/barapp/dto/ComandaDTO.java
package org.pbarreiro.barapp.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ComandaDTO {
    private UUID id;

    @NotNull(message = "La mesa es obligatoria")
    private Long mesaId;
    
    private MesaDTO mesa;

    @NotNull(message = "El camarero es obligatorio")
    private UUID camareroId;
    
    private PerfilDTO camarero;

    private OffsetDateTime fechaApertura;
    private OffsetDateTime fechaCierre;
    private Boolean estadoPago;
    private BigDecimal total;
    
    private List<DetalleComandaDTO> detalles;
}

File: mobile_client/BarAPP/composeApp/src/commonMain/kotlin/org/pabarreiro/barapp/data/remote/dto/ComandaDTO.kt
package org.pabarreiro.barapp.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class ComandaDTO(
    val id: String? = null,
    val mesaId: Long,
    val camareroId: String,
    val fechaApertura: String? = null,
    val fechaCierre: String? = null,
    val estadoPago: Boolean? = false,
    val total: Double? = 0.0,
    val detalles: List<DetalleComandaDTO>? = emptyList()
)
